### PR TITLE
Add meson

### PR DESCRIPTION
**Pull request by jmillikin**

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (meson) have been updated in this PR. Please review the changes.

**Comment by aignas** [comment]
Out of interest, what would be the benefit of doing it this way rather than through pip.parse and requirements files? Who would be the meson consumer in this case? Is it rules_foreign_cc?

**Comment by jmillikin** [comment]
Meson is designed to run directly from its source archive. It doesn't have any dependencies on PyPI packages. It also doesn't need to be installed via `setup.py` / `requirements.txt` / `pyproject.toml` / whatever the current situation in Python packaging looks like. The only dependency it's pulling from `rules_python` is the Python interpreter.

The consumer would be a build rule that's wrapping an external project's Meson-based build system. Whether that's a custom-written rule (I have a lot of those...) or something like `rules_foreign_cc` depends on the complexity of the external project being built.

`rules_foreign_cc` could use this module to obtain Meson if it wanted to, but it might not be worth the effort to them. Their Meson-related code has a lot of logic for interop with C/C++ ecosystem tooling like CMake or GNU Autotools. The part of their repo that this new module corresponds most closely to is only about ~50 lines in their external dependencies file: [`toolchains/built_toolchains.bzl#L206-L251`](https://github.com/bazelbuild/rules_foreign_cc/blob/0.12.0/toolchains/built_toolchains.bzl#L206-L251).

**Comment by aignas** [comment]
Thanks for explaining this. This was not a blocking comment and I just wanted to know more how rules_python is used.

File: modules/meson/metadata.json
Comment (Wyverald) [inline modules/meson/metadata.json:5]: would you mind adding yourself as a maintainer? it should be fairly low-maintenance.

**Review by Wyverald — APPROVED** [review]

File: modules/meson/metadata.json
Comment (jmillikin) [reply modules/meson/metadata.json]: I've done so to unblock, but in general I'd prefer to not be registered as the maintainer of other people's projects. I have no special knowledge about the internals of Meson.

**Review by fmeum — APPROVED** [review]

File: modules/meson/metadata.json
Comment (fmeum) [reply modules/meson/metadata.json]: Being a module maintainer only means that you will be the point of contact for the BCR entry (e.g. if someone else submits a new version), not upstream.

**Pull request merged**
